Python/chapter03/p01_three_in_one/miguelHx.py

```diff
@@ -0,0 +1,420 @@
+"""
+Python version 3.7.0
+3.1 - Three in one
+Describe how you could use a single array to implement three stacks
+"""
+
+import copy
+import unittest
+from typing import List
+
+
+class StackNode(object):
+
+    def __init__(self, data):
+        self.data = data
+        self.next = None # next is a pointer to a StackNode object
+
+class MyStack(object):
+    """
+    Stack data structure implementation.
+    Uses LIFO (last-in first-out) ordering.
+    The most recent item added to the stack is
+    the first removed.  Traversal is top to bottom.
+    """
+
+    def __init__(self):
+        self.top = None # top is a pointer to StackNode object
```

Comment:
```suggestion
        self.top: Optional[StackNode[T]] = None
```

it might be a good idea to use [type annotations](https://docs.python.org/3/library/typing.html) to convey this information, and might as well add them to all the public interface. it helps both humans (to quickly understand what's going on) and machines (with [mypy](http://mypy-lang.org/), code can be super quickly validated for type-correctness, which catches some families of bugs at "compile time")! static analysis FTW!

Reply:
Fixed in b84c365dc814a76fb065081d91b604bab2eb32a6